engine/runtime-compiler/src/main/java/org/enso/compiler/pass/MiniPassTraverser.java

```diff
@@ -76,16 +77,48 @@ static IR compileDeep(IR root, MiniIRPass miniPass) {
   private static List<IR> enqueueSubExpressions(
       Collection<MiniPassTraverser> queue, IR ir, MiniIRPass miniPass) {
     var childExpressions = new ArrayList<IR>();
-    var i = new int[1];
+    var i = new AtomicInteger();
+    if (ir instanceof Module mod) {
+      mod.imports()
+          .foreach(
+              v -> {
+                iterateSubExpressions(i, childExpressions, queue, v, miniPass);
+                return null;
+              });
+      mod.exports()
+          .foreach(
+              v -> {
+                iterateSubExpressions(i, childExpressions, queue, v, miniPass);
+                return null;
+              });
+      mod.bindings()
+          .foreach(
```

Comment:
The problem is that this has to be 1:1 change. E.g. one cannot change the number of bindings, imports or exports...